### PR TITLE
Add the ability to get the difficulty per-world

### PR DESCRIPTION
**Pull request by jacklin213**
The return type can be subject to change depending on how sponge handles difficulty

### CODE REVIEW

**Comment by ryantheleach** [comment]
You could just have this like gamemodes, but make it implement Compareable so you can rank any added difficulty levels.

**Comment by Zidane** [comment]
I need this in an enum-like list unless you choose not to @jacklin213 

If so, we'll need to delegate this to a team member and close.

**Comment by jacklin213** [comment]
@Zidane So return a list of game modes?

File: src/main/java/org/spongepowered/api/world/World.java [hunk 1]
```diff
@@ -40,6 +40,13 @@
     UUID getUniqueID();
 
     /**
+     * Gets the difficulty setting for this world.
+     * 
+     * @return difficulty of world
```
Comment (me4502) [inline src/main/java/org/spongepowered/api/world/World.java]: Yeah, capitalisation here is still missing.

**Comment by Bammerbom** [comment]
How about setting the difficulty per world?

**Comment by jacklin213** [comment]
@Bammerbom ah right i need to add that in, @turt2live I'll change that while im at it as well

**Comment by Aaron1011** [comment]
There should also be getter methods in `GameRegistry`

**Comment by jacklin213** [comment]
Not sure if the javadoc was done properly on the last commit, can someone go over the "ies" for me please

**Comment by kenzierocks** [comment]
I would just stick with the plural form.

File: src/main/java/org/spongepowered/api/world/World.java [hunk 2]
```diff
@@ -51,6 +51,20 @@
     UUID getUniqueID();
 
     /**
+     * Gets the {{@link Difficulty}} setting for this world.
```
Comment (kenzierocks) [inline src/main/java/org/spongepowered/api/world/World.java]: I don't think you want two {{ here.

**Comment by simon816** [comment]
How about getting the default difficulty defined in the server.properties file?
Something like `Server#getDifficulty()`

Also, this has merge conflicts - needs fixing.

**Comment by jacklin213** [comment]
@simon816 Then wouldn't it make sense to also have a ```Server#setDifficulty()```

**Comment by gratimax** [comment]
Pinging @SpongePowered/developers one last time before merge -- I'll probably also squash some smaller commits

File: src/main/java/org/spongepowered/api/GameRegistry.java [hunk 3]
```diff
@@ -550,4 +551,10 @@
      */
     RecipeRegistry getRecipeRegistry();
 
+	/**
```
Comment (Minecrell) [inline src/main/java/org/spongepowered/api/GameRegistry.java]: The indentation here is weird

**Pull request closed without merging**